include/onnxruntime/core/platform/ort_mutex.h

```diff
@@ -187,3 +188,35 @@ std::cv_status OrtCondVar::wait_for(std::unique_lock<OrtMutex>& cond_mutex,
 }
 };  // namespace onnxruntime
 #endif
+
+namespace onnxruntime {
+
+/*
+OrtSpinLock implemented mutex semantic "lock-freely",
+calling thread will not be put to sleep on blocked,
+which reduces cpu usage on context switching.
+*/
+struct OrtSpinLock {
```

Comment:
Can you put this in a separate file? may be ort_spin_lock.h. Also, it needs to be guarded by the new macro. Regular builds don't need this.

Reply:
The spin lock has potentially be consumed by other structures and functions.
I want to make it be public and easy to use.

Reply:
Ok, can you put it in a separate file?